### ISSUE TITLE
Move entry point for setting stream parameters into param

### ISSUE THREAD

**Issue by jlstevens** [holoviz/holoviews#1719]
The streams system currently has well defined semantics regarding parameters: mutation to parameters is only allowed via the ``event`` method. This is to ensure that parameter changes are triggering subscribers to reflect the updates and implies all stream parameters should be readonly to prevent them being mutated by other means.

Although there are unit tests to check that all the stream parameters directly offered by HoloViews are readonly, it is a pain for users to declare all the parameters as readonly in their custom stream classes (which they should!). If readonly is strictly enforced, using ``self`` in methods is less of a hazard as there is no chance that the parameter state will be changed.

The nicest solution long term is to guarantee that *all* parameter changes to stream classes *always* trigger subscribers instead of have to declare them as readonly and then requiring parameters are only set via ``event``.

In other words, it would be nice to automatically hook into params own parameter setting machinery to ensure the subscribers are called. Then when you set a parameter explicitly e.g ``stream_class.xpos=3``, you could be sure that the corresponding visualization would update without having to call ``event``. This might require a metaclass but definitely requires that param allow access to the param setting machinery.

**Comment by philippjfr** [comment]
This has now been addressed using the ``Params`` stream and the param.depends decorator. It should still be documented but I don't think we now have to make ``Stream`` objects be reactive.

**Comment by philippjfr** [comment]
We have used and documented support for parameters in a number of places throughout the docs now.
